src/tsd-metrics.ts

```diff
@@ -0,0 +1,359 @@
+import TsdMetricsList = metricsList.TsdMetricsList;
+import TsdMetricsEvent = metricsEvent.TsdMetricsEvent;
+import TsdMetricsFactory = metricsFactory.TsdMetricsFactory
+
+/**
+ * Node JS class for publishing metrics as time series data (TSD).
+ *
+ * @class
+ * @author Mohammed Kamel (mkamel at groupon dot com)
+ */
+export class TsdMetrics implements tsdDef.Metrics {
+
+    /**
+     * Metrics state object that holds flag to the metrics open state along with common event emitter.
+     */
+    private _metricsStateObject:MetricsStateObject = new MetricsStateObject();
+    private _timers:{[name:string]: tsdDef.Timer} = {};
+    private _counters:{[name:string]: tsdDef.Counter} = {};
+    private _sinks:tsdDef.Sink[];
+
+    /**
+     * MetricsEvent field that will be used to serialize the final metric event.
+     */
+    private _metricsEvent:tsdDef.MetricsEvent = new TsdMetricsEvent();
+
+    private static getOrCreateMetricsCollection<T>(name:string, collectionOfT:{[name:string]:T}, factory:() => T) {
+        if (collectionOfT[name] === undefined) {
+            collectionOfT[name] = factory();
+        }
+        return collectionOfT[name];
+    }
+
+    private getOrCreateCounterSamples(name:string):CounterSamples {
+        return <CounterSamples>TsdMetrics.getOrCreateMetricsCollection(
+            name, this._metricsEvent.counters, () => new CounterSamples(name, this._metricsStateObject));
+    }
+
+    private getOrCreateTimerSamples(name:string):TimerSamples {
+        return <TimerSamples>TsdMetrics.getOrCreateMetricsCollection(
+            name, this._metricsEvent.timers, () => new TimerSamples(name, this._metricsStateObject));
+    }
+
+    private getOrCreateGaugesSamples(name:string):tsdDef.MetricsList < tsdDef.MetricSample > {
+        return TsdMetrics.getOrCreateMetricsCollection(
+            name, this._metricsEvent.gauges, () => new TsdMetricsList < tsdDef.MetricSample >());
+    }
+
+    /**
+     * Constructor.
+     */
+    constructor(serviceName:string, clusterName:string, hostName:string, sinks:tsdDef.Sink[]) {
+        this._sinks = sinks;
+        this.annotate("_service", serviceName);
+        this.annotate("_cluster", clusterName);
+        this.annotate("_host", hostName);
+        this.annotate("_start", new Date().toISOString());
+    }
+
+    /**
+     * Create and initialize a counter sample. It is valid to create multiple
+     * <code>Counter</code> instances with the same name, even concurrently,
+     * each will record a unique sample for the counter of the specified name.
+     *
+     * @method
+     * @param {string} name The name of the counter.
+     * @returns {Counter} TsdCounter instance for recording a counter sample.
+     */
+    public createCounter(name:string):tsdDef.Counter {
+        if (this._metricsStateObject.assertIsOpen()) {
+            return this.getOrCreateCounterSamples(name).addCounter();
+        }
+        return new TsdCounter(name, this._metricsStateObject);
```

Comment:
This instance never does - if the metrics instance is closed (thereby failing the if check on line 112) then this timer simply provides a functioning interface so as to not cause exceptions, but ultimately never gets logged anywhere.